cpp/arcticdb/column_store/column_data.cpp

```diff
@@ -0,0 +1,18 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#include "column_data.hpp"
+
+namespace arcticdb {
```

Comment:
There are a lot of things in the ColumnData class that can be moved to a `.cpp` file. I didn't want to move all of them in this PR because it would have become harder to review. That's why I added a file only for the new things and we can add the rest in a separate PR.